packages/components/src/search-control/index.native.js

```diff
@@ -1,96 +1,237 @@
 /**
  * External dependencies
  */
-import { TextInput, View, TouchableHighlight } from 'react-native';
+import {
+	TextInput,
+	Text,
+	View,
+	TouchableOpacity,
+	Platform,
+	useColorScheme,
+} from 'react-native';
 
 /**
  * WordPress dependencies
  */
-import { useState, useRef } from '@wordpress/element';
-import { usePreferredColorSchemeStyle } from '@wordpress/compose';
+import { useState, useRef, useMemo, useEffect } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import { ToolbarButton } from '@wordpress/components';
+import { Button, Gridicons } from '@wordpress/components';
 import {
 	Icon,
-	cancelCircleFilled,
-	arrowLeft,
-	search as searchIcon,
+	cancelCircleFilled as cancelCircleFilledIcon,
+	arrowLeft as arrowLeftIcon,
+	close as closeIcon,
 } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
-import styles from './style.scss';
+import allStyles from './style.scss';
+import platformStyles from './platform-style.scss';
+
+// Merge platform specific styles with the default styles.
+const baseStyles = { ...allStyles };
```

Comment:
@fluiddot  I just went with a spread here which I know will only do a one level deep copy. I'm not sure if our style setup allows sass type nesting (now or in the future). My feeling is that this should be safe enough, thoughts?

Reply:
The StyleSheet object of React native only admits one level so it should be safe 👍 .